src/harness/audio/miniaudio.c

```diff
@@ -182,8 +182,12 @@ tAudioBackend_error_code AudioBackend_SetVolume(void* type_struct_sample, int vo
     miniaudio = (tMiniaudio_sample*)type_struct_sample;
     assert(miniaudio != NULL);
 
-    // convert from directsound -10000 - 0 decibel volume scale
-    linear_volume = ma_volume_db_to_linear(volume_db / 100.0f);
+    // convert from Carmageddon "decibel" scale to linear <0.0f..1.0f>
+    linear_volume = (volume_db + 350.0f) / -5.0f;
```

Comment:
this is one of these tricky parts where we are currently correctly emulating the behavior of the win95 directsound implementation. 

However, the win95 code was ported from the dos version with some unintended changes, and we are ok with changing the s3 code to more closely match the DOS version instead of the win95 version.

For example, https://github.com/dethrace-labs/dethrace/blob/main/src/S3/s3sound.c#L365-L368 does not appear in the DOS version at all. If it helps to remove it and just pass the raw `total_vol` to `AudioBackend_SetVolume`, that would be ok too, if it makes the resulting audio match up better with the dos behavior

Reply:
What about https://github.com/dethrace-labs/dethrace/blob/main/src/S3/s3sound.c#L361-L363?

Does it exist in DOS version? It looks like avoidance of "divide by zero" in [L365-L368](https://github.com/dethrace-labs/dethrace/blob/main/src/S3/s3sound.c#L365-L368) and might not be needed too.

Nonetheless, if L365-L368 is removed, `total_vol` is in range <0..510> and would have to be scaled down to <0.0f..1.0f> either in `miniaudio.c` or `s3sound.c`. I guess a preference is to do it `miniaudio.c`, right?

Reply:
Yes, in miniaudio.c